File: solutions_5658571765186560_0/Python/alkjash/p4.py
#! /usr/bin/env python

import sys
fin = open(sys.argv[1], "r")
fout = open("p4.out", "w")

T = int(fin.readline())

for i in xrange(T):
    X, R, C = map(int, fin.readline().split())
    R, C = sorted((R, C))    # R <= C
    ans = "GABRIEL"
    if 2 * R < X:            # L won't fit
        ans = "RICHARD"
    elif C < X:              # I won't fit
        ans = "RICHARD"
    elif ((R * C) % X) != 0: # Not divisible
        ans = "RICHARD"
    elif (X, R) in [(4, 2), (6, 3)]: # T tetrominoes
        ans = "RICHARD"
    elif X >= 7:             # Holes
        ans = "RICHARD"

    if ans not in ["GABRIEL", "RICHARD"]:
        print "Error!"
    fout.write("Case #" + str(i+1) + ": " + ans + "\n")

File: solutions_5658571765186560_0/C++/goodideax/p4.cpp
#include <iostream>
#include <string>
using namespace std;
int main()
{
	int n;
	int X,R,C;
	cin>>n;
	for(int i=0;i<n;i++)
	{
		int flag = 0;
		cin>>X>>R>>C;
		if(X==1) flag =1;
		if(X==2&&(R*C)%2==0&&R*C>=2) flag=1;
		if(X==3&&(R*C)%3==0&&R!=1&&C!=1)
			flag=1;
		if(X==4&&(R*C)%4==0&&R*C>=12&&R>2&&C>2)
			flag = 1;
		
		if(flag)
			cout<<"Case #"<<i+1<<": GABRIEL"<<endl;
		else
			cout<<"Case #"<<i+1<<": RICHARD"<<endl;
	}
	return 0;
}
